File: OnlineSurveyTool.Server.DAL.Models/Question.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace OnlineSurveyTool.Server.DAL.Models
{
    [Table("Questions")]
    public class Question : EntityBaseStringId
    {
        //Order inside survey
        [Required]
        public int Number { get; set; }

        [Required]
        public string Value { get; set; }

        [Required]
        [ForeignKey("Survey")]
        public string SurveyId { get; set; }

        [Required]
        public QuestionType Type { get; set; }

        //If Type is SingleChoice these properties do not mean anything
        //If Type is MultipleChoice these properties mean minimal and maximal number of choices a user can mark
        //If Type is NumericalInteger or NumericalDouble then these specify minimal and maximal value of an answer
        //If Type is Textual these specify minimal and maximal length of an answer string
        public double? Minimum { get; set; }

        public double? Maximum { get; set; }

        [Required]
        public bool CanBeSkipped { get; set; }

        //This is null for Numerical and Textual type of questions
        public virtual ICollection<ChoiceOption> ChoiceOptions { get; set; }

        public virtual Survey Survey { get; set; }
    }
}

File: onlinesurveytool.client/src/app/models/question.ts
import {ChoiceOption} from "./choice.option";

export enum QuestionType {
  SingleChoice = 'Single Choice',
  MultipleChoice = 'Multiple Choice',
  NumericalInteger = 'Numerical Integer',
  NumericalDouble = 'Numerical Double',
  Textual = 'Textual'
}

export interface Question {
    number: number;
    value: string;
    type: QuestionType;
    canBeSkipped: boolean;
    id?: string;
    minimum?: number;
    maximum?: number;
    choiceOptions?: Array<ChoiceOption>
}

export function questionPrototype(type: QuestionType, number: number = 0, value: string = ''): Question{
  switch(type) {
    case QuestionType.SingleChoice:
      return {
        number: number,
        value: value,
        type: QuestionType.SingleChoice,
        canBeSkipped: true,
        choiceOptions: []
      }

    case QuestionType.MultipleChoice:
      return {
        number: number,
        value: value,
        type: QuestionType.SingleChoice,
        canBeSkipped: true,
        choiceOptions: [],
        minimum: 0,
        maximum: 0
    }

    case QuestionType.NumericalDouble:
      return {
        number: number,
        value: value,
        type: QuestionType.NumericalDouble,
        canBeSkipped: true,
        minimum: 0,
        maximum: 100
      }

    case QuestionType.NumericalInteger:
      return {
        number: number,
        value: value,
        type: QuestionType.NumericalInteger,
        canBeSkipped: true,
        minimum: 0,
        maximum: 100
      }

    case QuestionType.Textual:
      return {
        number: number,
        value: value,
        type: QuestionType.Textual,
        canBeSkipped: true,
        minimum: 0,
        maximum: 100
      }
  }
}
